### PR TITLE
Add pyogrio

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Hmm, on linux, there are conflicts in solving the build environment: `Examining conflict for numpy libgdal python pip cython setuptools: ...`

(https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=199625&view=logs&jobId=10bda42a-d420-5c28-a43f-bc4e66405e5c&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0)

**Comment by jorisvandenbossche** [comment]
@ocefpaf any idea why I get conflicts in solving the conda environment ("Unsatisfiable dependencies for platform linux-64:")? It doesn't look different from eg fiona (which also includes numpy and libgdal)

**Comment by ocefpaf** [comment]
> @ocefpaf any idea why I get conflicts in solving the conda environment ("Unsatisfiable dependencies for platform linux-64:")? It doesn't look different from eg fiona (which also includes numpy and libgdal)

@jorisvandenbossche the max Python version in staged-recipes is 3.7 as far as I know. Let me check something first. Pinging @isuruf who probably knows better.

File: recipes/pyogrio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+source:
+  # no official release yet, for now use archive fallback
+  url: https://github.com/brendan-ward/pyogrio/archive/3e5174ab8250c4b1b0c84e0c8fbb916f9f839564.zip
+  sha256: 025adc6703602826ec8529a3de581c9735b9539ddba5d9d39cf5042b8d33efa9
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [py<38 or win or python_impl == 'pypy']
```
Comment (ocefpaf) [inline recipes/pyogrio/meta.yaml:16]: When I bumped my local docker image from 3.7 to 3.8 this built without a problem. Is the minimum really 3.8? Just curious on what features you are using that are not on 3.7.

File: recipes/pyogrio/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - pip
+    - cython
+    - numpy 1.16
```
Comment (ocefpaf) [inline recipes/pyogrio/meta.yaml:26]: It is a mistake in staged-recipes to try numpy 1.9. Our global pin is 1.16.

File: recipes/pyogrio/meta.yaml (hunk 1 above)
Comment (jorisvandenbossche) [reply recipes/pyogrio/meta.yaml:16]: >  Is the minimum really 3.8? Just curious on what features you are using that are not on 3.7.

I suppose it's compatible with py 3.6/3.7 as well, but for now only tested on py 3.8. 
But, the original PR build it for all python versions, but was also failing for py3.7. So limiting to py3.8 today was more as an attempt to get it passing ;)

**Comment by jorisvandenbossche** [comment]
@ocefpaf thanks for looking into it!

**Comment by ocefpaf** [comment]
Do you want to make a stable release or do you want to publish this in a dev/rc channel?

**Comment by jorisvandenbossche** [comment]
Just asked @brendan-ward about it!

File: recipes/pyogrio/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipes/pyogrio/meta.yaml:16]: Nah. That was a red-herring. The problem was the numpy version in staged-recipes.

**Comment by jorisvandenbossche** [comment]
@ocefpaf Thanks for the merge!

**Pull request merged**